pkg/crc/api/handlers.go

```diff
@@ -30,15 +30,28 @@ func stopHandler(client machine.Client, crcConfig crcConfig.Storage, _ json.RawM
 	return encodeStructToJSON(commandResult)
 }
 
-func startHandler(client machine.Client, crcConfig crcConfig.Storage, _ json.RawMessage) string {
+func startHandler(client machine.Client, crcConfig crcConfig.Storage, args json.RawMessage) string {
 	startConfig := machine.StartConfig{
-		Name:          constants.DefaultName,
-		BundlePath:    crcConfig.Get(config.Bundle).AsString(),
-		Memory:        crcConfig.Get(config.Memory).AsInt(),
-		CPUs:          crcConfig.Get(config.CPUs).AsInt(),
-		NameServer:    crcConfig.Get(config.NameServer).AsString(),
-		GetPullSecret: getPullSecretFileContent(crcConfig),
-		Debug:         true,
+		Name:       constants.DefaultName,
+		BundlePath: crcConfig.Get(config.Bundle).AsString(),
+		Memory:     crcConfig.Get(config.Memory).AsInt(),
+		CPUs:       crcConfig.Get(config.CPUs).AsInt(),
+		NameServer: crcConfig.Get(config.NameServer).AsString(),
+		Debug:      true,
+	}
+	if args != nil {
+		a := startArgs{}
+		_ = json.Unmarshal(args, &a)
+		if a == (startArgs{}) {
```

Comment:
It's better to check `json.Unmarshal` error. You can write:

```
var parsedArguments startArgs
if err := json.Unmarshal(args, &a); err != nil {
  // do something
}
if parsedArguments.PullSecretFile == "" {
   // do something
}
```

Reply:
The error from unmarshal is only generated if its provided invalid json or not a pointer, which is not that useful. I'd still need to check if args is empty to bypass this unmarshalling part.